platform-services.md

```diff
@@ -0,0 +1,215 @@
+file handles message passing between iOS and the Flutter app, while the [`LocationProvider.m`](https://github.com/flutter/flutter/blob/master/examples/hello_services/ios/HelloServices/LocationProvider.m)
+file implements the code that queries the iOS system location API
+([`CLLocationManager`](https://developer.apple.com/library/ios/documentation/CoreLocation/Reference/CLLocationManager_Class/)).
+
+* `ios/Flutter/Generated.xcconfig`: Contains the configuration settings to build your iOS app. See
+the `hello_services` [README](https://github.com/flutter/flutter/blob/master/examples/hello_services/README.md)
+to learn more about the properties you can configure.
+
+* [`lib/main.dart`](https://github.com/flutter/flutter/blob/master/examples/hello_services/lib/main.dart):
+The main entry point for your Flutter app. This source file is written in Dart syntax. The main
+function uses the [`HostMessages`](http://docs.flutter.io/flutter/services/HostMessages-class.html)class,
+which provides a high-level abstraction for passing messages to “hosts” of platform-specific
+services.
+
+## Add a service to your app {#add-a-service-to-your-app}
+
+In your Dart code (`lib/main.dart` file):
+
+1. Define a method to send a request to your target service (e.g. `_getLocation()`). You must encode
+the request as a JSON message. You’ll typically invoke this method in a
+[`State`](http://docs.flutter.io/flutter/material/State-class.html) object, when the user performs
+an operation on the UI.
+2. Define a messageHandler to process the messages returned by the target service
+(e.g. `handleGetRandom()`), then add this handler to the
+[`HostMessages`](http://docs.flutter.io/flutter/services/HostMessages-class.html) object.
+
+In your iOS code:
+
+1. In Xcode, open the `HelloServices` iOS project.
+2. Implement a service provider (e.g. `LocationProvider`). The provider class must conform to the
+`FlutterMessageListener` protocol (specifically, you must implement the
+`(NSString*)didReceiveString:(NSString*)` function). If your app wants to connect asynchronously to
+the service, use the [`FlutterAsyncMessageListener`](https://github.com/flutter/engine/blob/master/sky/shell/platform/ios/framework/Headers/FlutterAsyncMessageListener.h)
+class. Your provider is where you put the business logic for your service calls.
+3. In your [`AppDelegate`](https://github.com/flutter/flutter/blob/master/examples/hello_services/ios/HelloServices/AppDelegate.m),
+initialize the service provider you defined earlier then add a message listener to that provider.
+
+In your Android code:
+
+1. In your main activity (e.g.
+[`ExampleActivity.java`](https://github.com/flutter/flutter/blob/master/examples/hello_services/android/app/src/main/java/com/example/flutter/ExampleActivity.java)),
+add a message listener to listen for your service calls.
+2. Add the code to marshal and unmarshal the JSON messages for your service calls.
+3. Add code to handle the business logic for your service calls.
```

Comment:
Again, this step doesn't exist in the iOS instructions.  The two sets of instructions should be exactly parallel because they're describing a very similar sequence of steps in two different settings.